### PR TITLE
Adds wildcardPolicy field to IngressController API

### PR DESCRIPTION
**Pull request by danehans**
In support of the [Wildcard Admission Policy](https://github.com/openshift/enhancements/pull/252) enhancement proposal.

/assign @ironcladlou @knobunc 
/cc @Miciah @frobware

### CODE REVIEW

File: operator/v1/types_ingress.go [hunk 1]
```diff
@@ -335,8 +335,36 @@ type RouteAdmissionPolicy struct {
...
 	NamespaceOwnership NamespaceOwnershipCheck `json:"namespaceOwnership,omitempty"`
+	// wildcardPolicy describes how routes with wildcard policies should be handled
+	// for the ingress controller. WildcardPolicy works in conjunction with the
+	// wildcard policy of routes [1] claimed by the ingress controller.
+	//
+	// [1] https://github.com/openshift/api/blob/master/route/v1/types.go
+	//
+	// If WildcardPolicy is updated from WildcardsAllowed to WildcardsDisallowed,
+	// routes admitted by the ingress controller with a wildcard policy of
+	// Subdomain must be updated to a wildcard policy of None to be readmitted.
```
Comment (ironcladlou) [inline operator/v1/types_ingress.go]: What about adding a stronger warning about newly invalidated routes experiencing immediate disruption?

**Comment by ironcladlou** [comment]
/approve

This lgtm, will let others take a look and tag

File: operator/v1/types_ingress.go (hunk 1 above)
Comment (danehans) [reply operator/v1/types_ingress.go]: @ironcladlou I updated the message to draw more attention to the caveat.

**Comment by ironcladlou** [comment]
I want to unblock you and we can make any docs adjustments in followups.

/lgtm

File: operator/v1/types_ingress.go [hunk 2]
```diff
@@ -335,8 +335,38 @@ type RouteAdmissionPolicy struct {
 	// If empty, the default is Strict.
 	// +optional
 	NamespaceOwnership NamespaceOwnershipCheck `json:"namespaceOwnership,omitempty"`
+	// wildcardPolicy describes how routes with wildcard policies should
+	// be handled for the ingress controller. WildcardPolicy controls use
+	// of routes [1] exposed by the ingress controller based on the route's
+	// wildcard policy.
+	//
+	// [1] https://github.com/openshift/api/blob/master/route/v1/types.go
+	//
```
Comment (Miciah) [inline operator/v1/types_ingress.go]: Please enumerate the possible values here.  (The note below mentions the two possible values, but it does not state that there are no other possible values, and anyway, an explicit enumeration is clearest.)

Comment (danehans) [reply operator/v1/types_ingress.go]: @Miciah good point. I updated the godocs to list the supported values.

**Pull request merged**